e2e/ruler_test.go

```diff
@@ -610,6 +638,22 @@ func testRulesGeneration(ctx context.Context, restConfig *rest.Config, kubeClien
 				return false, fmt.Errorf("get ConfigMap: %w", err)
 			}
 			data := cm.Data
+			if features.Config.Compression == monitoringv1.CompressionGzip {
+				// When compression is enabled, we expect the config map with recording rules to
+				// be compressed with gzip. Uncompressed all files for validation purposes.
+				data = map[string]string{}
+				for fn, comprData := range cm.BinaryData {
+					r, err := gzip.NewReader(bytes.NewReader(comprData))
+					if err != nil {
+						t.Fatal(err)
+					}
+					uncompressed, err := io.ReadAll(r)
+					if err != nil {
+						t.Fatal(err)
+					}
+					data[fn] = string(uncompressed)
+				}
+			}
 
 			// The operator observes Rules across all namespaces. For the purpose of this test we
```

Comment:
question: This seems risky, as we could end up deleting things we shouldn't have written to the configmap - can you explain why this is here?

Reply:
The explanation is that this is not my code so I didn't think much of it. It comes from the initial commit (see [main_test.go](https://github.com/GoogleCloudPlatform/prometheus-engine/commit/5498f6b1f8c6f7627ee3c72731b367e5b9cb2ac0)).

However, you make a great point so I removed that piece of code and consolidated this method with the other create rules method in this file to ensure that `data = want`.